plugins/modules/foreman_hostgroup.py

```diff
@@ -303,10 +358,42 @@ def main():
             entity_dict['config_groups'] = module.find_resources_by_name('config_groups', entity_dict['config_groups'], failsafe=False, thin=True)
 
         if 'puppet_proxy' in entity_dict:
-            entity_dict['puppet_proxy'] = module.find_resource_by_name('smart_proxies', name=entity_dict['puppet_proxy'], failsafe=False, thin=True)
+            entity_dict['puppet_proxy'] = get_smart_proxy(module, smart_proxies, entity_dict['puppet_proxy'])
 
         if 'puppet_ca_proxy' in entity_dict:
-            entity_dict['puppet_ca_proxy'] = module.find_resource_by_name('smart_proxies', name=entity_dict['puppet_ca_proxy'], failsafe=False, thin=True)
+            entity_dict['puppet_ca_proxy'] = get_smart_proxy(module, smart_proxies, entity_dict['puppet_ca_proxy'])
+
+        if 'openscap_proxy' in entity_dict:
+            entity_dict['openscap_proxy'] = module.find_resource_by_name('smart_proxies', name=entity_dict['openscap_proxy'], failsafe=False, thin=True)
+
+        if 'organization' in entity_dict:
+            if 'organizations' in entity_dict:
+                if entity_dict['organization'] not in entity_dict['organizations']:
+                    entity_dict['organizations'].append(entity_dict['organization'])
+            else:
+                entity_dict['organizations'] = [entity_dict['organization']]
+            entity_dict['organization'] = module.find_resource_by_name('organizations', name=entity_dict['organization'], failsafe=False, thin=True)
+            organization = entity_dict['organization']
+
+        if 'organizations' in entity_dict:
+            entity_dict['organizations'] = module.find_resources_by_name('organizations', entity_dict['organizations'], thin=True)
+
+        if 'content_source' in entity_dict:
+            if not organization:
+                module.fail_json(msg="Please specify the organization.")
+            entity_dict['content_source'] = get_smart_proxy(module, smart_proxies, entity_dict['content_source'])
+
+        if 'lifecycle_environment' in entity_dict:
+            if not organization:
+                module.fail_json(msg="Please specify the organization.")
+            entity_dict['lifecycle_environment'] = module.find_katello_resource_by_name('lifecycle_environments', organization,
+                                                                                        name=entity_dict['lifecycle_environment'], failsafe=False, thin=True)
+
+        if 'content_view' in entity_dict:
+            if not organization:
```

Comment:
and here also https://github.com/theforeman/foreman-ansible-modules/pull/389/files#r312440972 ;)